### PR TITLE
[issue-787] added device parameter to the vision checks context

### LINKED ISSUE #787: [FEAT][CV] Add support for GPU objects
**Is your feature request related to a problem? Please describe.**
Current vision code supports only cpu objects, and even if user supplies a GPU model or data it'll crash.

**Describe the solution you'd like**
Need a global object (context) to recognize both if GPU is available and if the user objects are GPU, and adjust the rest of the code to cope with that.

### PR DESCRIPTION
**Pull request by yromanyshyn**
resolves #787

### CODE REVIEW

**Comment by ItayGabbay** [comment]
@yromanyshyn can you add tests?

**Review by noamzbr — COMMENTED** [review]
Two things:
1. Need to merge that with the recent refactor in vision base
2. Did we test that this work on gpu?

**Comment by yromanyshyn** [comment]
@noamzbr 

>Two things:
> Need to merge that with the recent refactor in vision base
> Did we test that this work on gpu?

1. of course, currently I am writing tests for base vision classes and would like to add them to the current PR, in the near time I will push that changes.
2. I tried to run it on the collab and checks worked/ but that probably is not a sufficient enough test, will think about how to do it better.

File: tests/vision/vision_conftest.py [hunk 1]
```diff
@@ -25,6 +26,12 @@
 )
 
 
+CURRENT_DIR = pathlib.Path(__file__).absolute().parent
+
+
+torch.hub.set_dir(str(CURRENT_DIR))
+
+
```
Comment (ItayGabbay) [inline tests/vision/vision_conftest.py:34]: This downgrades the code coverage.

File: deepchecks/vision/base.py [hunk 2]
```diff
@@ -287,14 +311,22 @@ def run(
...
+            train_dataset,
+            test_dataset,
+            model,
+            scorers=scorers,
+            scorers_per_class=scorers_per_class,
+            device=device
+        )
 
         # Create instances of SingleDatasetCheck for train and test if train and test exist.
         # This is needed because in the vision package checks update their internal state with update, so it will be
```
Comment (ItayGabbay) [inline deepchecks/vision/base.py:332]: What about adding the apply_to_tensor in the loop of the suite?

**Review by ItayGabbay — CHANGES_REQUESTED** [review]

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**